classes/task/get_queue_merging_requests.php

```diff
@@ -0,0 +1,81 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+/**
+ * Merge user accounts
+ * @package   tool_mergeusers
+ * @author    Nicola Vallinoto <n.vallinoto@liguriadigitale.it>
+ * @copyright 2023 Liguria Digitale
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace tool_mergeusers\task;
+use \tool_mergeusers\merge_request;
+/**
+ * Get queue merging requests task class
+ * @package   tool_mergeusers
+ * @author    Nicola Vallinoto <n.vallinoto@liguriadigitale.it>
+ * @copyright 2023 Liguria Digitale 
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class get_queue_merging_requests extends \core\task\scheduled_task {
+    public function get_name() {
+        return get_string('get_queue_merging_requests','tool_mergeusers');
+    }
+    /**
+     * Run task for getting the merging requests and adding them to adhoc task.
+     */
+    public function execute() {
+        mtrace("Task get_queue_merging_requests started.");
+        global $DB;
+        global $CFG;
+        // Read from moodle table records with status = QUEUED_NOT_PROCESSED.
+        // Add each record to adhoc task. 
+        // Update STATUS of each record added to adhoc task.
+        $notscheduled = merge_request::QUEUED_NOT_PROCESSED; 
+        $tablerequests = merge_request::TABLE_MERGE_REQUEST;      
+        $recordsnotyetscheduled = $DB->get_records($tablerequests, 
+                                                    ['status' => $notscheduled]);
+        foreach ($recordsnotyetscheduled as $record) {
+            // Add to adhoc_task - Create the instance.   
+            $mergerequestid = $record->id;
+            $mytask = new \tool_mergeusers\task\merge_user_accounts();
+            $mytask->set_custom_data(['mergerequestid' => $mergerequestid]);
+            // Queue the task.
+            \core\task\manager::queue_adhoc_task($mytask);
+            // Update the status of the tasked request.
+            $updated = merge_request::QUEUED_TO_BE_PROCESSED;
+            $this->update_status_table($mergerequestid,
+                                        $updated);
+            mtrace("User to remove: ".$record->removeuservalue." - User to keep: ".$record->keepuservalue);
+            mtrace("Adhoc task: merge request n. ".$mergerequestid." queued.");                            
+        }
+        mtrace("Task get_queue_merging_requests completed!");
+    }
+    /**
```

Comment:
sorry Jordi which is the convention ?

Reply:
Like [here](https://github.com/jpahullo/moodle-tool_mergeusers/pull/242/files#diff-eed3d8cd0b6673cd0efe1ae4ab6508b5c293194e78baeea59ef6377832ad81ceR51-R59), but with description on every `@param` informing what they are or what they content.

Reply:
In fact, it should be placed in any new method we are adding. If you use the local_codechecker plugin, I think it details you what to do. And if you use vscode as IDE, I suppose it would generate all the phpdoc properly. You just need to add complimentary details.

Reply:
in this case execute function has no parameters.